codegen/index.js

```diff
@@ -30,29 +31,26 @@ const types = {
   "selected": "Boolean",
   "size": "Number",
   "span": "Number",
-  "start": "Number",
-  "zoomAndPan": "String"
+  "start": "Number"
 };
+const reserved = ["module", "data", "type", "newtype", "class", "instance", "where", "derive", "if", "then", "else", "case", "of"];
 
-printRecord = (elProps) => `
+printRecord = (elProps) => elProps.length ? `
   ( ${ elProps.map((p) =>
-       `${p} :: ${types[p] || 'String'}`).join('\n  , ')
+       `${p} :: ${types[p] || "String"}`).join("\n  , ")
      }
-  )`
+  )` : "()"
 
 props.elements.html
-  .map((e) =>`
-    type Props_${e} = ${
-      props[e]
-        ? printRecord(props[e])
-        : '()'
-    }
+  .map((e) => `
+    type Props_${e} = ${printRecord(
+      (voids.includes(e) ? [] : ["children"]).concat(props[e] || []).sort()
+    )}
 
-    ${e}
+    ${reserved.includes(e) ? `${e}_` : e}
       :: forall attrs attrs_
-       . Union attrs attrs_ (SharedProps Props_${e}))
+       . Union attrs attrs_ (SharedProps Props_${e})
       => Record attrs
-      -> Array JSX
       -> JSX
-    ${e} = ${voids.indexOf(e) >= 0 ? 'createElementNoChildren' : 'createElement'} "${e}"
-`).forEach((x) => console.log(x.replace(/^\n\ {4}/, '').replace(/\n\ {4}/g, '\n')))
+    ${reserved.includes(e) ? `${e}_` : e} = createElement (unsafeCoerce "${e}")
```

Comment:
The `unsafeCoerce` converts the DOM string into a `ReactComponent`, which they are as far as `createElement` is concerned.  DOM components are weird that way.